src/main/java/org/generationcp/middleware/dao/dms/DmsProjectDao.java

```diff
@@ -1400,6 +1403,201 @@ public DatasetDTO getDatasetOfSampleList(final Integer sampleListId) {
+		sql.append("     cvtermSeason.definition AS " + StudySearchFilter.SEASON + ", ");
+		sql.append("     pmain.project_id AS " + StudySearchFilter.TRIAL_DB_ID + ", ");
+		sql.append(" 	 pmain.name AS " + StudySearchFilter.TRIAL_NAME + ", ");
+		sql.append("     MAX(pmain.start_date) AS " + StudySearchFilter.START_DATE + ", ");
+		sql.append("     MAX(pmain.end_date) AS " + StudySearchFilter.END_DATE + ", ");
+		sql.append("     CASE WHEN pmain.deleted = 0 THEN 1 ELSE 0 END AS " + StudySearchFilter.ACTIVE + ", ");
+		sql.append("     location.locid AS " + StudySearchFilter.LOCATION_DB_ID + ", ");
+		sql.append("     location.lname AS " + StudySearchFilter.LOCATION_NAME + ", ");
+		sql.append("     wp.project_name AS " + StudySearchFilter.PROGRAM_NAME + ", ");
+		sql.append("     wp.project_uuid AS " + StudySearchFilter.PROGRAM_DB_ID + " ");
+		this.appendStudySummaryFromQuery(sql);
+		this.appendStudySearchFilter(sql, studySearchFilter);
+		sql.append(" GROUP BY geoloc.nd_geolocation_id ");
+		if (!StringUtils.isEmpty(studySearchFilter.getSortedRequest().getSortBy()) && StudySearchFilter.SORTABLE_FIELDS
+			.contains(studySearchFilter.getSortedRequest().getSortBy())) {
+			sql.append(
+				" ORDER BY " + studySearchFilter.getSortedRequest().getSortBy() + " " + (
+					studySearchFilter.getSortedRequest().getSortOrder() == null ? "ASC" :
+						studySearchFilter.getSortedRequest().getSortOrder()));
+		}
+		return sql.toString();
+	}
+
+	private void appendStudySummaryFromQuery(final StringBuilder sql) {
+		sql.append(" FROM ");
+		sql.append("     nd_geolocation geoloc ");
+		sql.append("         INNER JOIN ");
+		sql.append("     nd_experiment nde ON nde.nd_geolocation_id = geoloc.nd_geolocation_id ");
+		sql.append("         INNER JOIN ");
+		sql.append("     project proj ON proj.project_id = nde.project_id ");
+		sql.append("         INNER JOIN ");
+		sql.append("     project pmain ON pmain.project_id = proj.study_id ");
+		sql.append("         LEFT OUTER JOIN ");
+		sql.append("     nd_geolocationprop geoprop ON geoprop.nd_geolocation_id = geoloc.nd_geolocation_id ");
+		sql.append("         LEFT OUTER JOIN ");
+		sql.append("     projectprop pProp ON pmain.project_id = pProp.project_id ");
+		sql.append("         LEFT OUTER JOIN ");
+		sql.append("     study_type studyType ON studyType.study_type_id = pmain.study_type_id ");
+		sql.append("         LEFT OUTER JOIN ");
+		sql.append(
+			"     nd_geolocationprop geopropSeason ON geopropSeason.nd_geolocation_id = geoloc.nd_geolocation_id AND geopropSeason.type_id = "
+				+ TermId.SEASON_VAR.getId());
+		sql.append("         LEFT OUTER JOIN ");
+		sql.append(
+			"     nd_geolocationprop geopropLocation ON geopropLocation.nd_geolocation_id = geoloc.nd_geolocation_id AND geopropLocation.type_id = "
+				+ TermId.LOCATION_ID.getId());
+		sql.append("         LEFT OUTER JOIN ");
+		sql.append("     location ON geopropLocation.value = location.locid");
+		sql.append("         LEFT OUTER JOIN ");
```

Comment:
There's no need to join to `workbench.workbench_project.` You can filter by program_uuid using `pmain.program_uuid`

Reply:
Oh nvm, I saw that we need to display program name